### ISSUE TITLE
right align the choice question

### ISSUE THREAD

**Issue by Anurangi** [sdaps/sdaps#158]
I am trying to build a questionnaire compatible with the newest stable version of sdaps.But the text of the choice questions are right aligned and looking ugly.when trying to use keyword "align" the pdf is not generated.Please advice on this

 \section{TEACHING}

    % And a more compact way of doing it; similar to markgroup
    \begin{choicegroup}{Please answer the following questions}
      % We have to add the possible choices at the start.
      \choice{Strongly disagree}
      \choice{Disagree}
      \choice{Neutral}
      \choice{Agree}
	  \choice{Strongly agree}
	     % After that it is possible to add each question.
      \question{The Lecturer was punctual}
      \question{The intended learning outcomes for the course have been made clear}
      \question{The lecturer could be clearly heard and understood}
      \question{Writing on a black/white board and/or presented slides were clear and visible}
      \question{The lecturer presented materials in a well organized way,with clear explanations,at an appropriate speed}
      \question{Adequate information and resources were available on the LMS}
      \question{The lecturer encouraged students to actively participate in the lecture and ask questions}
    \end{choicegroup}
![screenshot from 2018-12-17 09-18-11](https://user-images.githubusercontent.com/45928730/50065707-ebfae780-01dd-11e9-8850-0f2d4ec8659b.png)

**Comment by benzea** [comment]
Yeah, we changed the default, because in many cases it results in better readability. My guess is that your case is primarily different because the checkboxes end up taking a lot of space due to the headings.

Right now there is no built-in way to to change the alignment. However, you should be able to normal LaTeX commands to change the alignment. Try putting a `\raggedleft` call right at the start or by using a `\begin{flushright}`/`\end{flushleft}` environment.

**Comment by benzea** [comment]
Btw. I would suggest that you use an `optiongroup` or pass the `singlechoice` argument. That will result in circular checkboxes and SDAPS will ensure that only one item was checked.

**Comment by Anurangi** [comment]
Hi \raggedleft only apply changes to the part \begin{choicegroup}{Please answer the following questions} resulting the same out put.can you advise me on putting \begin{flushright} / end{flushright} environment cause it results me errors without generating the pdf.

**Comment by Anurangi** [comment]
using flushleft environment worked.also I would like to ask suggestions to improve the questionnaire readability by look at my work as I'm very new to this.Also I want to you can modify defs.py to consider any box that has markings in it as “checked” .But I'm having a hard time to understand the exact place in the defs.py that need the correction to be done.Please help

\begin{document}
  % Everything you do should be done inside the questionnaire environment.
 
  % If you don't like the default text at the beginning of each questionnaire
  % you can remove it with the optional [noinfo] parameter for the environment 
  \begin{questionnaire}
    % There is a predefined "info" style to hilight some text.
 \begin{info}
      \textbf{Limit your writing to the size of the textbox.}
    \end{info}   
 
\section{COURSE DETAILS}

\textbox*{2cm}{Name of the lecturer, Course Code, Course Title and the Department}

\begin{multicols}{3}
\textbox*{0.6cm}{Level}
\textbox*{0.6cm}{Semester}
\textbox*{0.6cm}{Date} 

 
 \end{multicols}

	
 \section{TEACHING}

    % And a more compact way of doing it; similar to markgroup
    \begin{choicegroup}[singlechoice]{Please answer the following questions}
      % We have to add the possible choices at the start.
      \choice{\tiny Strongly disagree}
      \choice{\tiny Disagree}
      \choice{\tiny Neutral}
      \choice{\tiny Agree}
	  \choice{\tiny Strongly agree}
	     % After that it is possible to add each question.    
      \question{\begin{flushleft}The Lecturer was punctual\end{flushleft}}
      \question{\begin{flushleft}The intended learning outcomes for the course have been made clear\end{flushleft}}
      \question{\begin{flushleft}The lecturer could be clearly heard and understood \end{flushleft}}
      \question{\begin{flushleft}Writing on a black/white board and/or presented slides were clear and visible \end{flushleft}}
      \question{\begin{flushleft}The lecturer presented materials in a well organized way,with clear explanations,at an appropriate speed\end{flushleft}}
      \question{\begin{flushleft}Adequate information and resources were available on the LMS \end{flushleft}}
      \question{\begin{flushleft}The lecturer encouraged students to actively participate in the lecture and ask questions\end{flushleft}}
	  \question{\begin{flushleft}There were sufficient tutorials, problem sets, or other learning activities assigned to help me understand the material\end{flushleft}}
	  \question{\begin{flushleft}Sufficient time was allocated outside class hours to meet the lecturer, to ask questions and obtain clarifications\end{flushleft}}
	  \question{\begin{flushleft}The feedback provided on learning the material, e.g., via tutorials, mid-semester tests, returned assignments, etc., was adequate.\end{flushleft}}
	  
    \end{choicegroup}
      
   \section{FACILITIES}
    % And a more compact way of doing it; similar to markgroup
    \begin{choicegroup}[singlechoice]{Please answer the following questions}
      % We have to add the possible choices at the start.
      \groupaddchoice{\tiny Strongly disagree}
      \groupaddchoice{\tiny Disagree}
      \groupaddchoice{\tiny Neutral}
      \groupaddchoice{\tiny Agree}
	  \groupaddchoice{\tiny Strongly agree} 
   
 
	 \choiceline {\begin{flushleft}The university library has an adequate number and selection of reference materials\end{flushleft}}
	 \choiceline{\begin{flushleft}Lecture hall facilities (space, multi-media projector, sound system, seating, etc.) were sufficient for learning\end{flushleft}}
	 \choiceline {\begin{flushleft}The black/white board was sufficiently large and visible\end{flushleft}}
   \end{choicegroup} 
 \textbox*{3cm}{Do you have any comments on teaching or facilities?}
  \end{questionnaire}
\end{document}

**Comment by Anurangi** [comment]
And also adding [singlechoice] keyword resulted me in not generating the report

**Comment by benzea** [comment]
0. Maybe I'll look at it later, but it isn't high on my priority list :)
1. You should set "checkmode=check" in the documentclass arguments.
2. [singlechoice] and no report? sounds like a bug, did you get a backtrace (could you open a new issue for that?)

**Comment by Anurangi** [comment]
\begin{choicegroup}[singlechoice]{Please answer the following questions} 
Is this the correct way to use singlechoice argument?I didn't open a new issue because I wasn't sure about whether I have used it correctly.

**Comment by Anurangi** [comment]
\choiceline {\begin{flushleft}The university library has an adequate number and selection of reference materials\end{flushleft}}
 Eventhough using flushleft environment works this whole thing appears in the report(question and \begin{flushleft} \end{flushleft} so It did not solve the problem unfortunately.

**Comment by Anurangi** [comment]
The report is generated but without the question parts I have used singlechoice keyword.

**Comment by benzea** [comment]
Yes, the flushleft/flushright stuff will appear in the report. You can prevent this by separately specifying the text to use for the report. i.e. `\choiceline[text={The university library has an adequate number and selection of reference materials}]{\begin{flushleft}The university library has an adequate number and selection of reference materials\end{flushleft}}`

Well, it did compile correctly, so yeah, that sounds like a bug in the report generation where the case is simply not handled. Which I probably did not notice when implementing it, because it was hidden by another issue. :)

See also: https://sdaps.org/class-doc/sdapsclassic.html#choicequestion
